pkg/template/controller/readiness.go

```diff
@@ -146,22 +146,22 @@ func checkRouteReadiness(obj runtime.Object) (bool, bool, error) {
 // readinessCheckers maps GroupKinds to the appropriate function.  Note that in
 // some cases more than one GK maps to the same function.
 var readinessCheckers = map[schema.GroupKind]func(runtime.Object) (bool, bool, error){
-	buildapi.LegacyKind("Build"):           checkBuildReadiness,
-	buildapi.Kind("Build"):                 checkBuildReadiness,
-	apps.Kind("Deployment"):                checkDeploymentReadiness,
-	extensions.Kind("Deployment"):          checkDeploymentReadiness,
-	appsapi.LegacyKind("DeploymentConfig"): checkDeploymentConfigReadiness,
-	appsapi.Kind("DeploymentConfig"):       checkDeploymentConfigReadiness,
-	batch.Kind("Job"):                      checkJobReadiness,
-	apps.Kind("StatefulSet"):               checkStatefulSetReadiness,
-	routeapi.Kind("Route"):                 checkRouteReadiness,
-	routeapi.LegacyKind("Route"):           checkRouteReadiness,
+	buildapi.Kind("Build"):                checkBuildReadiness,
+	apps.Kind("Deployment"):               checkDeploymentReadiness,
+	extensions.Kind("Deployment"):         checkDeploymentReadiness,
+	appsapi.Kind("DeploymentConfig"):      checkDeploymentConfigReadiness,
+	batch.Kind("Job"):                     checkJobReadiness,
+	apps.Kind("StatefulSet"):              checkStatefulSetReadiness,
+	routeapi.Kind("Route"):                checkRouteReadiness,
+	{Group: "", Kind: "Build"}:            checkBuildReadiness,
```

Comment:
@bparees when are you going to stop supporting oapi?

Reply:
The moment someone tells me we're allowed to drop it.

Reply:
> The moment someone tells me we're allowed to drop it.

The process endpoint no longer returns old oapi resources.  Is the only use-case you have for this working with legacy masters and previously created things?  Could you simply create a migration command to be run during 3.10 and remove this code in 3.11?

Reply:
is it possible we could accomplish this via conversion logic so the storage migration would just cover it?  Adding a whole new migration command for a single release seems like overkill.  (I assume you're suggesting a new "oc adm migrate templateinstances" subcommand)

Reply:
k.  i'll take a look at it.

Reply:
PR for migration: https://github.com/openshift/origin/pull/19664